Dart/src/com/jetbrains/lang/dart/resolve/DartResolver.java

```diff
@@ -22,9 +30,36 @@
 public class DartResolver implements ResolveCache.AbstractResolver<DartReference, List<? extends PsiElement>> {
   public static final DartResolver INSTANCE = new DartResolver();
 
+  public static final boolean USE_SERVER = false;
+
   @Nullable
   @Override
   public List<? extends PsiElement> resolve(@NotNull DartReference reference, boolean incompleteCode) {
+    if (USE_SERVER) {
+      final PsiFile refPsiFile = reference.getContainingFile();
+      //if (!DartAnalysisServerAnnotator.serverReadyForRequest(refPsiFile)) {
+      //  return null;
+      //}
+      final VirtualFile refVirtualFile = refPsiFile.getVirtualFile();
+      if (refVirtualFile != null) {
+        if (isIdentifier(reference)) {
+          Project project = reference.getProject();
+          final int refOffset = reference.getTextOffset();
+          final String refPath = refVirtualFile.getPath();
+          final List<NavigationRegion> regions = DartAnalysisServerService.getInstance().getNavigation(refPath);
+          for (NavigationRegion region : regions) {
+            if (region.containsInclusive(refOffset)) {
```

Comment:
Yes, we should.
I'm moving in this direction.
With this change to the server https://github.com/dart-lang/sdk/commit/7e194d8fb5960f6249e87b5648dc67c1e928cc64 we can start doing this.